pace-util/pace/util/quantity.py

```diff
@@ -306,8 +307,22 @@ def __init__(
                     f"got {type(data)}"
                 )
         elif gt4py_backend is not None:
+            extra_dims = [i for i in dims if i not in constants.SPATIAL_DIMS]
+            if len(extra_dims) > 0 or not dims:
+                mask = None
+            else:
+                mask = tuple(
+                    [
+                        any(dim in coord_dims for dim in dims)
+                        for coord_dims in [
+                            constants.X_DIMS,
+                            constants.Y_DIMS,
+                            constants.Z_DIMS,
+                        ]
+                    ]
+                )
             self._storage, self._data = self._initialize_storage(
-                data, origin, gt4py_backend
+                data, origin, gt4py_backend, mask
```

Comment:
nit: just a little worried about passing as positional args when we get to more than 3 arguments.
```suggestion
                data, origin=origin, gt4py_backend=gt4py_backend, mask=mask
```